src/table_properties.rs

```diff
@@ -0,0 +1,233 @@
+                fixed_key_len: v.get_u64(DBTableProperty::FixedKeyLen),
+                column_family_id: v.get_u64(DBTableProperty::ColumnFamilyId),
+                column_family_name: try!(v.get_str(DBTableProperty::ColumnFamilyName)),
+                filter_policy_name: try!(v.get_str(DBTableProperty::FilterPolicyName)),
+                comparator_name: try!(v.get_str(DBTableProperty::ComparatorName)),
+                merge_operator_name: try!(v.get_str(DBTableProperty::MergeOperatorName)),
+                prefix_extractor_name: try!(v.get_str(DBTableProperty::PrefixExtractorName)),
+                property_collectors_names:
+                    try!(v.get_str(DBTableProperty::PropertyCollectorsNames)),
+                compression_name: try!(v.get_str(DBTableProperty::CompressionName)),
+                user_collected_properties: HashMap::new(),
+            };
+            let mut user_it = UserCollectedPropertiesIter::new(v.get_user_properties());
+            while user_it.valid() {
+                {
+                    let k = user_it.key();
+                    let v = user_it.value();
+                    props.user_collected_properties.insert(k.to_owned(), v.to_owned());
+                }
+                user_it.next();
+            }
+            collection.insert(k, props);
+            it.next();
+        }
+        Ok(collection)
+    }
+}
+
+struct TablePropertiesCollectionIter {
+    pub inner: *mut DBTablePropertiesCollectionIterator,
+}
+
+impl Drop for TablePropertiesCollectionIter {
+    fn drop(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_table_properties_collection_iter_destroy(self.inner);
+        }
+    }
+}
+
+impl TablePropertiesCollectionIter {
+    fn new(inner: *mut DBTablePropertiesCollection) -> TablePropertiesCollectionIter {
+        unsafe {
+            TablePropertiesCollectionIter {
+                inner: crocksdb_ffi::crocksdb_table_properties_collection_iter_create(inner),
+            }
+        }
+    }
+
+    fn valid(&self) -> bool {
+        unsafe { crocksdb_ffi::crocksdb_table_properties_collection_iter_valid(self.inner) }
+    }
+
+    fn next(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_table_properties_collection_iter_next(self.inner);
+        }
+    }
+
+    fn key(&self) -> Result<String, String> {
+        unsafe {
+            let mut klen: size_t = 0;
+            let k = crocksdb_ffi::crocksdb_table_properties_collection_iter_key(
+                self.inner, &mut klen as *mut size_t);
+            let bytes = slice::from_raw_parts(k, klen);
+            String::from_utf8(bytes.to_owned()).or_else(|e| Err(format!("{}", e)))
+        }
+    }
+
+    fn value(&self) -> *mut DBTableProperties {
+        unsafe { crocksdb_ffi::crocksdb_table_properties_collection_iter_value(self.inner) }
+    }
+}
+
+struct UserCollectedPropertiesIter {
+    pub inner: *mut DBUserCollectedPropertiesIterator,
```

Comment:
Missing lifetime  constraint?
make it cannot outlive  UserCollectedProperties.
A TODO is ok.